storage/client_merge_test.go

```diff
@@ -227,6 +227,25 @@ func TestStoreRangeMergeNonConsecutive(t *testing.T) {
 		log.Errorf("split ranges keys are equal %q!=%q", rangeA.Desc().StartKey, rangeC.Desc().StartKey)
 	}
 
+	// Read all of the keys touched by the split transactions to resolve
+	// intents.  If intents are left unresolved then the asynchronous
+	// resolution may happen after the call to RemoveRange below,
+	// reviving the range and breaking the test.
+	// Unfortunately we must list those keys explicitly, since there are
+	// encoding problems with using Scan with system keys (#437).
+	// TODO(bdarnell): when we can use Scan with system keys, do so.
```

Comment:
PTAL; I changed it to use Scan. The problem I had was that I was using KeyMin, which includes local keys.

I'm not sure whether we'd be able to get rid of this when we change this test to use rebalancing. Range GC is going to be asynchronous in the real world because any message can revive a range; we may need to continue to forcibly resolve intents in tests that require range removal to be deterministic.